File: StringAverage.js
function averageString(str) {
  // Code away
  var int = new Object();
  int["zero"] = 0;
  int["one"] = 1;
  int["two"] = 2;
  int["three"] = 3;
  int["four"] = 4;
  int["five"] = 5;
  int["six"] = 6;
  int["seven"] = 7;
  int["eight"] = 8;
  int["nine"] = 9;
  int[0] = "zero";
  int[1] = "one";
  int[2] = "two";
  int[3] = "three";
  int[4] = "four";
  int[5] = "five";
  int[6] = "six";
  int[7] = "seven";
  int[8] = "eight";
  int[9] = "nine";
  
  var words = str.split(" ");
  var total = 0;
  for (var i = 0; i < words.length; i++) {
    var x = (int[words[i]]);
    total = total + x;
  }
  var avg = Math.floor(total/words.length)
  var result = int[avg]
  if (result == null || result > 9) {
    return "n/a"
  }else {
    return result
    }
}

File: Kata/Kata/StringAverage.cs
//using System;
//using System.Collections.Generic;
//using System.Linq;
//using System.Runtime.InteropServices;
//using System.Runtime.Remoting.Messaging;
//using System.Text;
//using System.Threading.Tasks;

//namespace Kata
//{
//    public class Kata
//    {
//        public static string AverageString(string str)
//        {
//            string[] words = str.Split(' ');
//            int total = 0;
//            for (var i = 0; i < words.Length; i++)
//            {
//                if (!System.Enum.IsDefined(typeof(EnumAverage), words[i]))
//                    return "n/a";
//                else
//                    foreach (string name in System.Enum.GetNames(typeof(EnumAverage)))
//                    {
//                        string strName = System.Enum.Parse(typeof(EnumAverage), name).ToString();
//                        if (words[i] == strName)
//                        {
//                            int strValue = (int) System.Enum.Parse(typeof(EnumAverage), strName);
//                            total = total + strValue;
//                        }
//                    }
//            }
//            int avg = (total / words.Length);
//            EnumAverage enumName = (EnumAverage)avg;
//            string avgName = enumName.ToString();
//            return avgName != "" || avg < 9 ? avgName : "n/a";
//        }

//        public static string AverageStringLinq(string str)
//        {
//            if (string.IsNullOrEmpty(str)) return "n/a";

//            var numbers = str.Split(' ').ToList();
//            if( numbers.Exists(s => !Enum.IsDefined(typeof(EnumAverage), s))) return "n/a";

//            return ((EnumAverage) (int) numbers.Select(s => (int) Enum.Parse(typeof(EnumAverage), s)).Average()).ToString();

//        }

//        private static bool IsValidInput(string input)
//        {
//            if (string.IsNullOrEmpty(input)) return false;

//            return true;
//        }

//        enum EnumAverage
//        {
//            zero,
//            one,
//            two,
//            three,
//            four,
//            five,
//            six,
//            seven,
//            eight,
//            nine,
//            ten
//        };
//    }
//}
